tests/php/test_class.jetpack_photon.php

```diff
@@ -229,6 +229,19 @@ public function test_photon_parse_images_from_html_a_tags_without_images() {
 		$this->assertEquals( $expected, print_r( Jetpack_Photon::parse_images_from_html( $sample_html ), true ) );
 	}
 
+	/**
+	 * Tests Photon's HTML parsing.
+	 *
+	 * @author biskobe
+	 * @covers Jetpack_Photon::filter_the_content
+	 * @since 3.2
+	 */
+	public function test_photon_parse_images_from_html_a_tags_with_hash_in_href() {
+		list( $sample_html, $expected ) = $this->get_photon_sample_content( 'a-tags-with-hash-href.html' );
+
+		$this->assertEquals( trim( $expected ), trim( Jetpack_Photon::filter_the_content( $sample_html ) ) );
```

Comment:
`trim()` because IDE wants to add a new line at the end of the file, which gets added to the expected results :/